### DISCUSSION TITLE
What affects Fan control CPU resources

### DISCUSSION THREAD

**Discussion by EdwardKiefer** [Rem0o/FanControl.Releases#939] [Q&A]
First I want to say I am really impressed with Fan control's low CPU usage, especially when compared to full HW monitoring programs like Libre hardware monitor and HWinfo64.

That said I do have some questions and a few suggestions with regard to CPU usage.

1) does having a lot of curves add to usage?
2) also what about curves that are connected/linked to a control card (device) but that device is disabled (not using fan control, still using bios).
3) Having many temp points on a custom curve?

It would be nice to know what parts affect CPU usage and by what degree roughly.
I don't know if your familiar with MSI AB or HWinfo64 they both have profiling polling data on each sensor type so you know which is high and can adjust/disable.

Since I end up using fan control for case fans with mix curves (GPU/CPU) and leave GPU and CPU fan to BIOSes control, I was thinking having a polling rate other than 1sec could trim some usage down. Since with a high response time don't really need fast polling.

Now here is one reason for my questions, when I first set up fan control I had it controlling my rear case fan and front with a mix of CPU and GPU curves I didn't control GPU or CPU fans. The usage was super low with an avg CPU of 0.01%, was like this for months and then I added CPU fan support just to see if improved anything. This was around ver 117-119 timeframe CPU avg usage went to 0.03% (which is still very low) but then I tried to go back to just case fan control and the avg usage stays around 0.02- 0.03 so trying to figure how to get back to extremely low usage in beginning.

I only use MB, CPU, GPU sensor enabled all rest disabled (storage, memory, PSU etc)

Oh just for comparison since those avg CPU% mean nothing to you here data from other programs (rough est)

MSI AB CPU avg usage 0.01% (note polling rate = 1750ms)
Libre HW monitor CPU avg usage 0.04% (note full sensor aray 1sec polling).

One last thing on Updater.exe does that only run once at startup or is it checking more often?

I hope this doesn't come off as a complaint, I just want to help improve the program as it is a unique program that bios and other programs don't address.

**Comment by Rem0o** [comment]
FanControl got profiled and optimized to death. You wouldn't be seeing sub 1% CPU usage otherwise.
Various things can increase CPU usage. As you guessed, the more things it manages, the more usage. That implies sensors and curves.  Also, if the UI is visible, more usage. 

One quick trick is to disable any sensor source that is not necessary in the Sensor settings window.  Also, if you have Nvidia GPU, use the NvAPIWrapper source instead of the GPU section of LHM, it's much more lightweight.

**Reply by EdwardKiefer** [reply]
Ok will try to streamline that sensor inputs, right now I only use CPU package temp and Nvidia GPU temp. I do have MB, GPU, CPU and NvAPwrapper enabled, so I can try disable a few things.
Maybe since I have libre HW monitor I will try there first to see which will disable ok and then move to Fan control just in case I disable something I am using so I don't need to resync, redo any setup.
Though I could make a new profile and mess with that.

Thanks for the tips.

**Reply by EdwardKiefer** [reply]
OK, I was able to disable CPU and GPU, I only had to change CPU package temp to CPU core temp which runs off MB sensors.

**Reply by EdwardKiefer** [reply]
Just an update, that worked and then some on.
Much lower now.
